### ISSUE TITLE
Http api calls with Digest Auth works only on iOs but not on Android.

### ISSUE THREAD

**Issue by TianYao226** [facebook/react-native#36017]
### Description

I am trying to make api calls using fetch, the end point structure is like this:
http://user:pass@${deviceIp}/api/audio-site/sites
Headers:
    const requestOptions = {
      method: 'GET',
      headers: {
          'content-type': 'application/json',
        },
      redirect: 'follow',
    };

So with postman I need to use Digest auth to pass in user and pass, then hit the endpoint.
So right now with iOs, the current code works fine by just sending in the url as it is, I guess bcoz the api can read the beginning of the url and separate user pass and host, from /api/audio-site/sites.
I am not sure. But on Android, this always returns me 404, not authorized. I also added android:usesCleartextTraffic="true" in my manifest.
I tried with react-native-digest-fetch library too but nothing works. Using this library I can see the logs that reads all the parameters I sent in correctly, and it did also the challenge and everything, but when it makes final fetch with the challenge, the reponse is always 404 and not json, but a text/html.
What is the reason that this code works on iOs but not on android? How can I solve it?

Thanks.

### Version

0.72.3

### Output of `npx react-native info`

ve-info
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/react-native-info - Not found
npm ERR! 404 
npm ERR! 404  'react-native-info@*' is not in this registry.
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/tiany/.npm/_logs/2023-01-31T12_59_56_506Z-debug-0.log

### Steps to reproduce

fetch

### Snack, code example, screenshot, or link to a repository

http://user:pass@${deviceIp}/api/audio-site/sites
Headers:
    const requestOptions = {
      method: 'GET',
      headers: {
          'content-type': 'application/json',
        },
      redirect: 'follow',
    };

**Comment by iBotPeaches** [comment]
>  What is the reason that this code works on iOs but not on android? How can I solve it?

@TianYao226 - It seems you are in the best position to figure that out. If you are logging the 404 and the output - what is it? Does that help piece it together?

You could experiment with attaching the digest as a header instead, which I vaguely remember is just `Authorization: Basic base64(user:pass)`

> 0.72.3

This isn't even in stable tree yet, so unsure what version you are actually on.

> npm ERR! code E404
> npm ERR! 404 Not Found - GET https://registry.npmjs.org/react-native-info - Not found

Additionally, this is clearly not the output intended. The command is `npx react-native info`. You had a hyphen between `react-native` and `info`.

**Comment by TianYao226** [comment]
Thank you for the answer. 
The response I got was a blob of unauthorized text.
I did try to use Authorization: "Digest" + user:pass, 
but nothing works. I also used the react-native-digest-fetch library, but it didnt work either.

when I used the diges fetch library it says:
parameters i sent in are:
parameters {"headers": {"content-type": "application/json", "redirect": "follow"}, "method": "GET", "password": "xxx", "username": "user"}

initialResults {"_bodyBlob": {"_data": {"__collector": [Object], "blobId": "d88d3152-3661-460c-8d16-a7c2e3fec673", "offset": 0, "size": 381}}, "_bodyInit": {"_data": {"__collector": [Object], "blobId": "d88d3152-3661-460c-8d16-a7c2e3fec673", "offset": 0, "size": 381}}, "bodyUsed": false, "headers": {"map": {"connection": "Keep-Alive", "content-length": "381", "content-type": "text/html; charset=iso-8859-1", "date": "Wed, 01 Feb 2023 09:07:41 GMT", "keep-alive": "timeout=5, max=100", "server": "Apache/2.4.54 (Unix) OpenSSL/1.1.1s", "www-authenticate": "Digest realm=\"AXIS_B8A44F23FCC2\", nonce=\"GPzQx5/zBQA=ce238e80b5d4f5b3fa7f06ad5f9efb2974b4148f\", algorithm=MD5, qop=\"auth\"", "x-content-type-options": "nosniff", "x-frame-options": "SAMEORIGIN", "x-xss-protection": "1; mode=block"}}, "ok": false, "status": 401, "statusText": "", "type": "default", "url": "http://user:pass@192.111.111.12/audio-site/v3.0/sites"}

I also see it went to get challenge, and do the hash of password etc. But my reponse is always unauthorized when it comes back to my app.

Yes I run npx react-native info again:

Binaries:
    Node: 19.4.0 - /usr/local/bin/node
    Yarn: Not Found
    npm: 9.2.0 - /usr/local/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.11.3 - /Users/tiany/.rvm/rubies/ruby-2.7.6/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK: Not Found
  IDEs:
    Android Studio: 2022.1 AI-221.6008.13.2211.9477386
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.18 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.1.0 => 18.1.0 
    react-native: 0.70.6 => 0.70.6 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
